File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function(nums) {
    /* takes more instructions then sort function
    for(let i=0; i<nums.length; i++){
        nums[i] = nums[i] * nums[i]
    }
    
    for(let i=0;i<nums.length-1;i++){
        for(let j=i+1;j<nums.length; j++){
            if(nums[i]> nums[j]){
                let temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
            }
        }
    }
    return nums;
    */

    
    let n = nums.length
    for(let i=0; i<n; i++){
        nums[i] = nums[i] **2;
    }
    nums.sort((x,y) => x-y);
    return nums;

};

File: Leet Code/0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int n = nums.length;
        int[] result = new int[n];
        int j =0;
        int k = n-1;
        for(int i=n-1; i>=0; i--){
            if(nums[j]* nums[j] >= nums[k]*nums[k]){
                result[i] = nums[j]*nums[j];
                j++;
            }else if(nums[k]*nums[k] > nums[j]*nums[j]){
                result[i] = nums[k]*nums[k];
                k--;
            }
        }
        return result;
    }
}